gramex/data.py

```diff
@@ -1394,15 +1402,24 @@ def _filter_mongodb_col(col, op, vals, meta_cols):
         return {col: {_mongodb_op_map[op]: convert(val)} for val in vals}
 
 
-def _mongodb_query(args, meta_cols):
+def _mongodb_query(args, meta_cols, results=None):
     # Convert a query like x>=3&x>=4&x>=5 into
     # {"$or": [{x: {$gt: 3}}, {x: {$gt: 4}}, {x: $gt: 5}]}
     # TODO: ?_id= is not working
+    import bson
+
+    object_key = None
+    if results:
+        for k, v in results[0].items():
+            if type(v) == bson.objectid.ObjectId:
+                object_key = k
```

Comment:
What if there are multiple object keys? Make this a set, not a single column name. Name it object_keys. Add a comment explaining that Pandas returns strings for both.

Another possibility. Don't use Pandas to read the first few rows. Instead, use pymongo & detect the type directly. This may be better.